zcash_primitives/src/sighash_versioning.rs

```diff
@@ -0,0 +1,112 @@
+//! Sighash versioning as specified in [ZIP-246].
+//!
+//! [ZIP-246]: https://zips.z.cash/zip-0246
+
+use alloc::{collections::BTreeMap, vec::Vec};
+use lazy_static::lazy_static;
+
+use orchard::{
+    issuance_sighash_versioning::IssueSighashVersion,
+    orchard_sighash_versioning::OrchardSighashVersion,
+};
+
+/// The sighash version and associated data
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SighashInfo {
+    version: u8,
+    associated_data: Vec<u8>,
+}
+
+impl SighashInfo {
+    /// Constructs a `SighashInfo` from raw bytes.
+    ///
+    /// Returns `None` if `bytes` is empty.
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        bytes.split_first().map(|(&version, info)| Self {
+            version,
+            associated_data: info.to_vec(),
+        })
+    }
+
+    /// Returns the raw bytes of the `SighashInfo`.
+    pub fn to_bytes(&self) -> Vec<u8> {
+        [vec![self.version], self.associated_data.clone()].concat()
+    }
+
+    #[cfg(zcash_unstable = "nu7")]
+    pub(crate) fn to_orchard_version(&self) -> Option<OrchardSighashVersion> {
+        ORCHARD_SIGHASH_VERSION_TO_INFO
+            .iter()
+            .find(|(_, v)| *v == self)
+            .map(|(k, _)| k.clone())
+    }
+
+    pub(crate) fn to_issuance_version(&self) -> Option<IssueSighashVersion> {
```

Comment:
this one is also `#[cfg(zcash_unstable = "nu7")]`

Reply:
I we add this flag, we have to add it to all issuance code.